libs/features/personalization/personalization.js

```diff
@@ -1145,14 +1160,29 @@ export async function init(enablements = {}) {
       experiments: [],
       geoPrefix: config.locale?.prefix.split('/')[1]?.toLowerCase() || 'en-us',
     };
-    manifests = manifests.concat(await combineMepSources(pzn, promo, mepParam));
-    manifests?.forEach((manifest) => {
-      if (manifest.disabled) return;
-      const normalizedURL = normalizePath(manifest.manifestPath);
-      loadLink(normalizedURL, { as: 'fetch', crossorigin: 'anonymous', rel: 'preload' });
-    });
-    if (pzn) loadLink(getXLGListURL(config), { as: 'fetch', crossorigin: 'anonymous', rel: 'preload' });
+    manifestPromises = manifestPromises
+      .concat(await combineMepSources(pzn, promo, promoUtilsPromise, mepParam));
+
+    manifestPromises = manifestPromises.map((mPromise) => mPromise.then(async (resp) => {
+      try {
+        if (!resp.ok) {
+          /* c8 ignore next 5 */
+          if (resp.status === 404) {
+            throw new Error('File not found');
+          }
+          throw new Error(`Invalid response: ${resp.status} ${resp.statusText}`);
+        }
+        const manifestData = await resp.json();
+        return { manifestData, manifestPath: resp.url };
+      } catch (e) {
+        /* c8 ignore next 3 */
+        console.log(`Error loading content: ${resp.url}`, e.message || e);
+      }
+      if (pzn) loadLink(getXLGListURL(config), { as: 'fetch', crossorigin: 'anonymous', rel: 'preload' });
```

Comment:
Doesn't this line still need to run even when the manifest loading is successful?